src/cryptonote_core/blockchain.cpp

```diff
@@ -5021,6 +5021,8 @@ bool Blockchain::prepare_handle_incoming_blocks(const std::vector<block_complete
         unsigned nblocks = batches;
         if (i < extra)
           ++nblocks;
+        if (!nblocks)
```

Comment:
If you're really testing against 0, wouldn't it be better to write:
`if (nblocks == 0)`?
Just an opinion.